src/pu-emmc.c

```diff
@@ -881,5 +896,55 @@ pu_emmc_new(const gchar *device_path,
     if (!pu_emmc_parse_clean(self, root, error))
         return NULL;
 
+    /* Check if raw binary overwrites partitions */
+    if (self->partitions && self->raw) {
+        PuEmmcPartition *part = self->partitions->data;
+        gsize part_start = part->offset * self->device->sector_size;
+        gsize raw_start;
+        gsize raw_end;
+        struct stat st;
+        for (GList *b = self->raw; b != NULL; b = b->next) {
+            PuEmmcBinary *bin = b->data;
+            g_autofree gchar *path = NULL;
+
+            path = pu_path_from_uri(bin->input->uri, prefix, error);
+            if (path == NULL) {
+                g_prefix_error(error, "Failed parsing input URI for partition: ");
+                return NULL;
+            }
+            if (stat(path, &st) == -1) {
```

Comment:
How would this look like if we were using GLib's `GFileInfo` and `g_file_info_get_size()`? There may be a little more code, but I would prefer using GLib's existing functions instead of using system libraries, for better portability.